recipe/get_plugins.py

```diff
@@ -47,42 +49,96 @@ def checksum(htname, fname):
 
 
 def unzip(fname, dname=None):
+    src_dname, ext = os.path.splitext(fname)
     if dname is None:
-        dname = os.path.splitext(fname)[0]
-
-    with zipfile.ZipFile(fname, "r") as fh:
-        fh.extractall(dname)
+        dname = src_dname
+
+    if ext == ".7z":
+        subprocess.check_call(["7za", "x", "-aos", "-o" + dname, "--", fname])
+    else:
+        with zipfile.ZipFile(fname, "r") as fh:
+            common_prefix_parts = os.path.commonprefix(fh.namelist()).split("/")
+            if len(common_prefix_parts) > 1:
+                # If everything is inside a single top-most folder (e.g., like in GitHub archives),
+                # skip recreating that part of the hierarchy.
+                root_folder = common_prefix_parts[0] + "/"
+                members = []
+                for zip_info in fh.infolist():
+                    new_file_name = zip_info.filename[len(root_folder):]
+                    if new_file_name:
+                        zip_info.filename = new_file_name
+                        members.append(zip_info)
+                fh.extractall(dname, members)
+            else:
+                fh.extractall(dname)
 
     os.remove(fname)
 
     return dname
 
 
-Download = collections.namedtuple("Download", ["url", "hashtype", "hashval"])
+def check_hash(src, fname):
+    hashval = checksum(src.hashtype, fname)
+    if src.hashval != hashval:
+        raise ValueError(
+            "Checksum mismatch in file '%s' %s != %s" % (fname, src.hashval, hashval)
+        )
+
+
+Download = collections.namedtuple("Download", ["url", "hashtype", "hashval", "dname"])
 
 urls = [
     Download(
-        url="http://nsis.sourceforge.net/mediawiki/images/8/8f/UAC.zip",
+        url="http://code.kliu.org/misc/elevate/elevate-1.3.0-redist.7z",
+        hashtype="sha256",
+        hashval="b1b3f070353a0eadee2cea3a575049d10df9763ff24e39313da4cec9455382e1",
+        dname="elevate"
+    ),
+    Download(
+        url="http://nsis.sourceforge.net/mediawiki/images/7/79/UAC_v0.2.2d.zip",
```

Comment:
Guessing this is the version Anaconda is using? Seems fine to me if so.

Reply:
Yes, that's the archive I found which had the `dll` with the same checksum as in Anaconda.